File: trunk/ShadowFrameworkLite20/src/shadow/pipeline/SFStructureStamp.java
package shadow.pipeline;

import java.util.Iterator;
import java.util.List;

import shadow.pipeline.parameters.SFParameteri;

public class SFStructureStamp {

	private int[] data;
	private int sum;

	public SFStructureStamp(SFPipelineStructure structure) {
		List<SFParameteri> parameters=structure.getAllParameters();
		this.data=new int[parameters.size()];
		int i=0;
		sum=0;
		for (Iterator<SFParameteri> iterator=parameters.iterator(); iterator
				.hasNext(); i++) {
			SFParameteri sfParameteri=(SFParameteri) iterator.next();
			short type=sfParameteri.getType();
			data[i]=SFParameteri.getTypeDimension(type);
			sum+=data[i];
		}
	}
	
	public int[] getData() {
		return data;
	}

	public int size() {
		return data.length;
	}

	public int getDimension(int index) {
		return data[index];
	}

	public int getSum() {
		return sum;
	}
	
}

File: ShadowFrameworkJS/src/shadow/pipeline/SFStructureStamp.js
//Java to JS on 21/03/2012

function SFStructureStamp(structure){
	var parameters=structure.getAllParameters();
	this.data=new Array();
	this.sum=0;
	for(var i=0;i<parameters.length;i++){
		sfParameteri=parameters[i];
		var type=sfParameteri.getType();
		this.data[i]=static_SFParameteri.getTypeDimension(type);
		this.sum+=data[i];
	}
}


SFStructureStamp.prototype["getData"]=function(){
			return this.data;
		};

SFStructureStamp.prototype["size"]=function(){
			return this.data.length;
		};

SFStructureStamp.prototype["getDimension"]=function(index){
			return this.data[index];
		};

SFStructureStamp.prototype["getSum"]=function(){
			return sum;
		};
